models/web/snowplow_media_interactions_this_run.sql

```diff
@@ -0,0 +1,65 @@
+{{
+  config(
+    materialized='table',
+    tags=["this_run"],
+    sort = 'start_tstamp',
+    dist = 'event_id'
+  )
+}}
+
+with prep as (
+
+ select
+    e.event_id,
+    e.page_view_id,
+    e.domain_sessionid,
+    e.domain_userid,
+    coalesce(y.player_id, me.html_id) as media_id,
+    {{ dbt_utils.surrogate_key(['e.page_view_id', 'coalesce(y.player_id, me.html_id)' ]) }} play_id,
+    round(mp.duration) as duration,
+    case when me.media_type = 'audio' then 'audio' else 'video' end as media_type,
+    coalesce(y.schema_vendor||'-'||y.schema_name, me.schema_vendor||'-'||me.schema_name)  as media_player_type,
+    e.page_referrer,
+    e.page_url,
+    coalesce(y.url, me.current_src) as source_url,
+    e.geo_region_name,
+    e.br_name,
+    e.dvce_type,
+    e.os_name,
+    e.os_timezone,
+    mpe.type as event_type,
+    e.derived_tstamp as start_tstamp,
+    mp.current_time as player_current_time,
+    isnull(mp.playback_rate, 1) as playback_rate,
+    coalesce(y.playback_quality, ve.video_height||'x'||ve.video_width) as playback_quality,
+    case when event_type = 'ended' then 100 else mp.percent_progress end percent_progress,
+    mp.muted as is_muted
+
+    from {{ ref("snowplow_web_base_events_this_run") }} as e
+
+    inner join {{ source('atomic', 'com_snowplowanalytics_snowplow_media_player_event_1') }} as mpe
+    on mpe.root_id = e.event_id
```

Comment:
You should be joining on the tstamps here too otherwise you risk introducing duplicates